landing-zone-submodule/management-vpc/main.tf

```diff
@@ -0,0 +1,61 @@
+##############################################################################
+# Resource Group
+##############################################################################
+
+module "resource_group" {
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-resource-group.git?ref=v1.0.5"
+  # if an existing resource group is not set (null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+#############################################################################
+# Provision cloud object storage and bucket
+#############################################################################
+
+resource "ibm_resource_instance" "cos_instance" {
+  count = var.enable_vpc_flow_logs ? 1 : 0
+
+  name              = "${var.prefix}-vpc-logs-cos"
+  resource_group_id = module.resource_group.resource_group_id
+  service           = "cloud-object-storage"
+  plan              = var.cos_plan
+  location          = var.cos_location
+}
+
+resource "ibm_cos_bucket" "cos_bucket" {
```

Comment:
I think this should use the cos module to create a cos instance and a cos bucket https://github.com/terraform-ibm-modules/terraform-ibm-cos . There is also a fscloud compliant cos coming at https://github.com/terraform-ibm-modules/terraform-ibm-cos/pull/279 - but for now you could just point to the root of the cos module.

Reply:
Thinking further, I think the cos bucket should always be passed as a reference (input) of the submodules. Otherwise we'll end up pulling too much from the landing zone module, eg key protect, etc

Reply:
Updated the code to get existing cos instance guid and bucket_name.